### ISSUE TITLE
SProp axiom bug

### ISSUE THREAD

**Issue by artagnon** [coq/coq#16427]
When printing axioms using SProp on Coq 8.15.1, we get:

```text
STrue relies on definitional UIP.
```

I think this is  a bug.

**Comment by SkySkimmer** [comment]
duplicate #16219

**Comment by artagnon** [comment]
Hi, this bug is not fixed by #16219. When I `Print Assumptions NTypes.` in [bonak](https://github.com/artagnon/bonak), I get:

```
Section Variables:
side : HSet
Axioms:
STrue relies on definitional UIP. (* bug? *)
functional_extensionality_dep
  : forall (A : Type) (B : A -> Type) (f g : forall x : A, B x),
	(forall x : A, f x = g x) -> f = g
fext_computes
  : forall (A : Type) (B : A -> HSet) (f : forall a : A, B a),
    functional_extensionality_dep f f (fun a : A => eq_refl) = eq_refl
```

**Comment by SkySkimmer** [comment]
Cannot reproduce. Are you sure you used the right Coq version?

**Comment by artagnon** [comment]
I'm using coq.8.16.1. I don't know if there's been a regression since 8.16.0, because bonak failed to build on 8.16.0.

**Comment by SkySkimmer** [comment]
The fix is in 8.17 not 8.16

**Comment by artagnon** [comment]
Thanks for clarifying. I thought the fix made it to 8.16.1.
